mitmproxy/addons/modifybody.py

```diff
@@ -0,0 +1,134 @@
+import os
+import re
+import typing
+
+from mitmproxy import exceptions
+from mitmproxy import flowfilter
+from mitmproxy import ctx
+from mitmproxy.utils import strutils
+
+
+def parse_modify_body(s):
```

Comment:
This looks suspiciously similar to `parse_modify_headers` now. We can probably just merge those two?

 - We can raise a plain `ValueError` here and re-raise an OptionsError in each addon with an individual error message.
 - Let's also include `strutils.escaped_str_to_bytes()` and `flowfilter.parse` here.

Reply:
Good idea. What's the preferred approach here? Have the function in one addon file and import it in the other?

Regarding the bullet points:
1) You mean a `ValueError` in the parse method and an `OptionsError` in the `Configure` method? Does that make sense here, i.e., having 2 subsequently thrown exceptions for the same error? (Trying to learn here, not to correct you :))
2) These methods are now applied in `replace()`. You suggest to move them to the parse method, right?

Reply:
> Have the function in one addon file and import it in the other?

yup, let's keep it simple. :)

> You mean a ValueError in the parse method and an OptionsError in the Configure method? Does that make sense here, i.e., having 2 subsequently thrown exceptions for the same error? (Trying to learn here, not to correct you :))

Yes. The very Pythonic way would be to re-raise as follows:

```python
def foo():
    raise ValueError("qux")

def bar():
    try:
        foo()
    except ValueError as e:
        raise BarError("cannot foo() today") from e
```

This ensure the following: 1) You can make sure that bar() only raises BarErrors. 2) You get a very nice chained traceback: 

```
Traceback (most recent call last):
  File "<pyshell#55>", line 3, in bar
    foo()
  File "<pyshell#49>", line 2, in foo
    raise ValueError("qux")
ValueError: qux

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<pyshell#56>", line 1, in <module>
    bar()
  File "<pyshell#55>", line 5, in bar
    raise BarError("cannot foo() today") from e
BarError: cannot foo() today
```

> These methods are now applied in replace(). You suggest to move them to the parse method, right?

Yes. Reading from file should stay inside replace() however, it can be very useful that file contents are not cached. :)

Reply:
Done. Please check.

I haven't moved `flowfilter.parse` to the `parse_modify_hook` method as then the thrown `ValueError` is ambigous, i.e., when executing `parse_modify_hook` again in the `configure` method to re-raise the exception, we'd have to check the error message to determine whether the `ValueError` was caused by an invalid "/.../.../..." option or by an invalid filter.
Hope this makes sense.